### PR TITLE
Add a Continue Reading Section

### PR DESCRIPTION
**Pull request by shortspider**
**Changes**

Added a Continue Reading section for books that are in progress.

**Issues**

### CODE REVIEW

**Comment by shortspider** [comment]
Hey will this be reviewed or did I miss a step?

File: src/components/homesections/homesections.js [hunk 1]
```diff
@@ -365,58 +369,10 @@ import ServerConnections from '../ServerConnections';
...
-
-    function loadResumeVideo(elem, apiClient) {
+    function loadResume(elem, apiClient, headerText, mediaType) {
         let html = '';
 
-        html += '<h2 class="sectionTitle sectionTitle-cards padded-left">' + globalize.translate('HeaderContinueWatching') + '</h2>';
+        html += '<h2 class="sectionTitle sectionTitle-cards padded-left">' + globalize.translate(headerText) + '</h2>';
         if (enableScrollX()) {
             html += '<div is="emby-scroller" class="padded-top-focusscale padded-bottom-focusscale" data-centerfocus="true">';
             html += '<div is="emby-itemscontainer" class="itemsContainer scrollSlider focuscontainer-x" data-monitor="videoplayback,markplayed">';
```
Comment (dmitrylyzo) [inline src/components/homesections/homesections.js:378]: Here and below `data-monitor` contains `videoplayback`. Audio equivalent contains `audioplayback`.

Comment (shortspider) [reply src/components/homesections/homesections.js]: Thanks for reviewing @dmitrylyzo and for catching this. I've fixed it.

File: src/components/homesections/homesections.js [hunk 2]
```diff
@@ -365,63 +369,23 @@ import ServerConnections from '../ServerConnections';
...
-            centerText: !cardLayout,
-            allowBottomPadding: false,
-            cardLayout: cardLayout,
-            showYear: true,
-            lines: 2
-        });
-    }
-
-    function loadResumeVideo(elem, apiClient) {
+    function loadResume(elem, apiClient, headerText, mediaType) {
```
Comment (dmitrylyzo) [inline src/components/homesections/homesections.js:372]:
```suggestion
    const dataMonitorHints = {
        'Audio': 'audioplayback,markplayed',
        'Video': 'videoplayback,markplayed'
    };

    function loadResume(elem, apiClient, headerText, mediaType) {
```

_Use `Add suggestion to batch` if you want a single commit._

File: src/components/homesections/homesections.js [hunk 3]
```diff
@@ -365,63 +369,23 @@ import ServerConnections from '../ServerConnections';
...
         let html = '';
 
-        html += '<h2 class="sectionTitle sectionTitle-cards padded-left">' + globalize.translate('HeaderContinueWatching') + '</h2>';
+        let dataMonitor = 'data-monitor="';
+        if (mediaType == 'Video') {
+            dataMonitor += 'videoplayback,'
+        } else if (mediaType == 'Audio') {
+            dataMonitor += 'audioplayback,'
+        }
+        dataMonitor += 'markplayed">';
```
Comment (dmitrylyzo) [inline src/components/homesections/homesections.js:381]:
```suggestion
        const dataMonitor = dataMonitorHints[mediaType] || 'markplayed';
```

File: src/components/homesections/homesections.js [hunk 4]
```diff
@@ -365,63 +369,23 @@ import ServerConnections from '../ServerConnections';
...
+
+        html += '<h2 class="sectionTitle sectionTitle-cards padded-left">' + globalize.translate(headerText) + '</h2>';
         if (enableScrollX()) {
             html += '<div is="emby-scroller" class="padded-top-focusscale padded-bottom-focusscale" data-centerfocus="true">';
-            html += '<div is="emby-itemscontainer" class="itemsContainer scrollSlider focuscontainer-x" data-monitor="videoplayback,markplayed">';
+            html += '<div is="emby-itemscontainer" class="itemsContainer scrollSlider focuscontainer-x" ' + dataMonitor;
```
Comment (dmitrylyzo) [inline src/components/homesections/homesections.js:386]:
```suggestion
            html += `<div is="emby-itemscontainer" class="itemsContainer scrollSlider focuscontainer-x" data-monitor="${dataMonitor}">`;
```

File: src/components/homesections/homesections.js [hunk 5]
```diff
@@ -365,63 +369,23 @@ import ServerConnections from '../ServerConnections';
...
         } else {
-            html += '<div is="emby-itemscontainer" class="itemsContainer padded-left padded-right vertical-wrap focuscontainer-x" data-monitor="videoplayback,markplayed">';
+            html += '<div is="emby-itemscontainer" class="itemsContainer padded-left padded-right vertical-wrap focuscontainer-x" '+ dataMonitor;
```
Comment (dmitrylyzo) [inline src/components/homesections/homesections.js:388]:
```suggestion
            html += `<div is="emby-itemscontainer" class="itemsContainer padded-left padded-right vertical-wrap focuscontainer-x" data-monitor="${dataMonitor}">`;
```

File: src/components/homesections/homesections.js [hunk 6]
```diff
@@ -142,15 +144,17 @@ import ServerConnections from '../ServerConnections';
...
         } else if (section === 'resumeaudio') {
-            loadResumeAudio(elem, apiClient);
+            return loadResume(elem, apiClient, 'HeaderContinueListening', 'Audio');
         } else if (section === 'activerecordings') {
             loadLatestLiveTvRecordings(elem, true, apiClient);
         } else if (section === 'nextup') {
             loadNextUp(elem, apiClient);
         } else if (section === 'onnow' || section === 'livetv') {
             return loadOnNow(elem, apiClient, user);
+        } else if (section == 'resumebook') {
```
Comment (dmitrylyzo) [inline src/components/homesections/homesections.js:156]:
```suggestion
        } else if (section === 'resumebook') {
```

**Review by dmitrylyzo — COMMENTED** [review]
Need to add `resumebook` option to all these selects (`Settings` -> `Home`): https://github.com/jellyfin/jellyfin-web/blob/ab9bc8185ffa721db26a49d091ba20a163510c6a/src/components/homeScreenSettings/homeScreenSettings.template.html#L21-L110

_The only problem - it is currently broken: #2630_

And on the server side: https://github.com/jellyfin/jellyfin/blob/master/Jellyfin.Data/Enums/HomeSectionType.cs
```cs
/// <summary>
/// Continue Reading.
/// </summary>
ResumeBook = 9
```

**Comment by shortspider** [comment]
> And on the server side: https://github.com/jellyfin/jellyfin/blob/master/Jellyfin.Data/Enums/HomeSectionType.cs

How do I go about updating the server side? Is that just a separate PR in that project? Do I need to link it to this in some way? Sorry I haven't done cross project work before.

**Comment by dmitrylyzo** [comment]
> How do I go about updating the server side? Is that just a separate PR in that project? Do I need to link it to this in some way?

You need to open PR in https://github.com/jellyfin/jellyfin and add a note here.

**Comment by shortspider** [comment]
OK created a PR for the enum change at jellyfin/jellyfin#5944. All changes suggested by @dmitrylyzo made.

**Review by dmitrylyzo — APPROVED** [review]

**Comment by shortspider** [comment]
@dmitrylyzo anything I can do to help get another approval for this request?

**Review by thornbill — APPROVED** [review]

**Pull request merged**